bases/shared/entrypoint.sh

```diff
@@ -51,14 +54,126 @@ mkdir -p /state/cores
 chmod a+rwx /state/cores
 echo "/state/cores/core.%e.%p.%h.%t" > /proc/sys/kernel/core_pattern
 
-ln -sf "$SLOGFILE" /state/slogfile_current.json
-
 # Copy a /config/network/$basename to $BOOTSTRAP_CONFIG
 resolved_config=$(echo "$BOOTSTRAP_CONFIG" | sed 's_@agoric_/usr/src/agoric-sdk/packages_g')
 resolved_basename=$(basename "$resolved_config")
 source_config="/config/network/$resolved_basename"
 test ! -e "$source_config" || cp "$source_config" "$resolved_config"
 
+backup_log_files_and_cleanup() {
+    SERVICE_ACCOUNT_JSON="/config/secrets/logs-backup.json"
+    if [ -f "$SERVICE_ACCOUNT_JSON" ]
+    then
+        BUCKET_NAME="agoric-chain-logs"
+        SCOPES="https://www.googleapis.com/auth/devstorage.read_write"
+        STATE_DIRECTORY_PATH=/state
+        TOKEN_URI="https://oauth2.googleapis.com/token"
+
+        base64url_encode() {
+            openssl base64 -e -A | \
+            tr '+/' '-_' | \
+            tr -d '='
+        }
+
+        CLIENT_EMAIL=$(
+            jq '.client_email' \
+            --raw-output < $SERVICE_ACCOUNT_JSON
+        )
+        PRIVATE_KEY=$(
+            jq '.private_key' \
+            --raw-output < $SERVICE_ACCOUNT_JSON | \
+            sed 's/\\n/\n/g'
+        )
+
+        HEADER='{"alg":"RS256","typ":"JWT"}'
+        ISSUER="$CLIENT_EMAIL"
+        AUDIENCE="$TOKEN_URI"
+        EXPIRATION=$(("$BOOT_TIME"+3600))
+        ISSUED_AT="$BOOT_TIME"
+
+        PAYLOAD=$(cat <<EOF
+{ "aud": "$AUDIENCE", "exp": $EXPIRATION, "iat": $ISSUED_AT, "iss": "$ISSUER", "scope": "$SCOPES" }
+EOF
+        )
+
+        HEADER_BASE64=$(echo -n "$HEADER" | base64url_encode)
+        PAYLOAD_BASE64=$(echo -n "$PAYLOAD" | base64url_encode)
+
+        SIGNATURE=$(
+            echo -n "${HEADER_BASE64}.${PAYLOAD_BASE64}" | \
+            openssl dgst -sha256 -sign <(echo -n "$PRIVATE_KEY") | \
+            base64url_encode
+        )
+
+        JWT="${HEADER_BASE64}.${PAYLOAD_BASE64}.${SIGNATURE}"
+
+        ACCESS_TOKEN=$(
+            curl $TOKEN_URI \
+            --data "grant_type=urn:ietf:params:oauth:grant-type:jwt-bearer&assertion=$JWT" \
+            --header "Content-Type: application/x-www-form-urlencoded" --silent --request POST | \
+            jq --raw-output '.access_token'
+        )
```

Comment:
Can we move this SSO RPC inside  `upload_and_remove_file` as well so that it also runs in background? We can add check inside `upload_and_remove_file` that if `ACCESS_TOKEN` exists then don't make RPC (as upload_and_remove_file is called for every file).

If you think this RPC is not adding any major overhead then it's fine to make it blocking/sync.

Reply:
This takes neglibible time so isn't needed